plugins/bgp/bgp-plugin.go

```diff
@@ -40,12 +63,128 @@ func (p *BgpPlugin) Init() error {
 	p.KVScheduler.RegisterKVDescriptor(gd)
 
 	// register descriptor for bgp peer config
+
 	pd := descriptor.NewPeerConfDescriptor(p.Log, p.BGPServer)
 	p.KVScheduler.RegisterKVDescriptor(pd)
 
+	p.watchCloser = make(chan string)
+	watcher := p.Deps.KVStore.NewWatcher(nodePrefix)
+	err := watcher.Watch(p.onChange, p.watchCloser, "")
+	if err != nil {
+		return err
+
+	}
 	return nil
 }
 func (p *BgpPlugin) Close() error {
 	log.Println("Closing BgpPlugin Application.")
 	return nil
 }
+
+//tutorial says keyval.protowatchresp but i couldnt find it
+//in the  cn-infra/db/keyval/proto_watcher_api.go in our vendor folder there isnt a ProtoWatchResp
+// but on the github for cn infra, there is one
+func (p *BgpPlugin) onChange(resp datasync.ProtoWatchResp) {
+	//key := resp.GetKey()
+
+	//Getting
+	value := new(vppnode.VppNode)
+	changeType := resp.GetChangeType()
+	if changeType == datasync.Delete {
+		if prevValExist, err := resp.GetPrevValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+		} else if !prevValExist {
+			logging.Errorf("Delete and Previous Value Does not Exist")
+		}
+	} else {
+		if err := resp.GetValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+		}
```

Comment:
If there is an error getting the value, you may as well return - value is invalid, you're better off not doing anything. We can talk about possible recovery strategies later